cmd/minikube/cmd/start.go

```diff
@@ -232,6 +235,15 @@ func runStart(cmd *cobra.Command, args []string) {
 			glog.Errorln("Error writing crictl config: ", err)
 		}
 	}
+	// default network plugin (cni)
+	if selectedContainerRuntime != "" {
```

Comment:
Rather than checking the string value, please  merge some of this logic into the new cruntime package. One possibility, would be to pass the flag values into cruntime.Config{}, and then implement methods for getting back the CNI info. 

I'd say just try to find the simplest/best way you to do this without doing string comparisons =)

Reply:
Fixed